starter-code/src/functions-and-arrays.js

```diff
@@ -98,3 +185,25 @@ var matrix = [
   [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
+
+function greatestProduct(matrix){
+  var product = 0;
+  size = matrix.length;
+  for (var y=0; y<size; y++){
+    for (var x=0; x<size-3; x++){
+      var smallArr = matrix[0].slice(x, x+3);
+      var newProd = prodArray(smallArr);
+      if (newProd > product){
+        product = newProd;
+      }
+    }
+  }
+}
```

Comment:
You got the logic for the horizontal numbers! The full correction of the exercice is here : 

```
function greatestProduct(m) {
  var horizontal, vertical, diagonal;
  var greatProduct = 0;

  for (var i=0, nRows = m.length; i < nRows - 1; i++) {
    for (var j=0, nCols = m[0].length; j < nCols - 1; j++) {
      // 4 horizontal elements
      if (j < nCols-3) {
        horizontal = m[i][j] * m[i][j+1]   * m[i][j+2]   * m[i][j+3];
      }

      // 4 vertical elements
      if (i < nRows-3) {
        vertical   = m[i][j] * m[i+1][j]   * m[i+2][j]   * m[i+3][j];
      }

      // 4 Diagonal elements
      if (i < nRows-3 && j < nCols-3) {
        diagonal   = m[i][j] * m[i+1][j+1] * m[i+2][j+2] * m[i+3][j+3];
      }

      greatProduct = Math.max(greatProduct, horizontal, vertical, diagonal);
    }
  }

  return greatProduct;
}
```
Its globally very good! You got the for loops logic and your code is very clean.
Keep up the good work!